REPO_LAYOUT.md

```diff
@@ -0,0 +1,88 @@
+# Repository layout overview
+
+This is a high level overview of how the repository is laid out to both aid in code investigation,
+as well as to clearly specify how extensions are added to the repository. The top level directories
+are:
+
+* [.circleci/](.circleci/): Configuration for [CircleCI](https://circleci.com/gh/envoyproxy).
+* [bazel/](bazel/): Configuration for Envoy's use of [Bazel](https://bazel.build/).
+* [ci/](ci/): Scripts used both during CI as well as to build Docker containers.
+* [configs/](configs/): Example Envoy configurations.
+* [docs/](docs/): Project level documentation is well as scripts for publishing final docs during
+  releases.
+* [examples/](examples/): Larger Envoy examples using Docker and Docker Compose.
+* [include/](include/): "Public" interface headers for "core" Envoy (not extensions). In general,
```

Comment:
this is a bit confusingly phrased, given it includes the interfaces for major extensions, i.e. tcp proxying, http stream etc.